back/lib/time_boundaries_parser.rb

```diff
@@ -7,8 +7,8 @@ def initialize(start_at, end_at)
   def parse
     timezone = AppConfiguration.instance.settings('core', 'timezone')
     platform_range = AppConfiguration.instance.created_at..Time.now.in_time_zone(timezone).end_of_day
-    start_range = @start_at&.to_date ? @start_at : platform_range.begin
-    end_range = @end_at&.to_date ? @end_at : platform_range.end
+    start_range = @start_at&.to_date ? @start_at.to_date : platform_range.begin
+    end_range = @end_at&.to_date ? @end_at.to_date : platform_range.end
```

Comment:
This was returning a `Date` most of the time, but a `string` in some cases. I figured this was probably not intentional so I changed it to always return a `Date`